posthog/apps.py

```diff
@@ -66,3 +66,16 @@ def ready(self):
             logger.warning("Skipping async migrations setup. This is unsafe in production!")
         else:
             setup_async_migrations()
+
+        try:
+            r = get_client()
+            lock_key = "posthog_sync_hog_function_templates_task_lock"
+            # setnx returns True if the key was set, False if it already exists
+            if r.setnx(lock_key, 1):
+                r.expire(lock_key, 60 * 60)  # expire after 1 hour
+                logger.info("Queuing sync_hog_function_templates celery task (redis lock)...")
+                sync_hog_function_templates_task.delay()
+            else:
+                logger.info("Not queuing sync_hog_function_templates task: lock already set")
```

Comment:
For tidiness sake can we put this in a function somewhere else (like next to the sync task code)